lib/ads/client_side_ad_manager.js

```diff
@@ -180,6 +183,16 @@ shaka.ads.ClientSideAdManager = class {
             this.video_.offsetHeight, viewMode);
       });
 
+      if ('ResizeObserver' in window) {
+        this.resizeObserver_ = new ResizeObserver(() => {
+          const viewMode = document.fullscreenElement ?
+              google.ima.ViewMode.FULLSCREEN : google.ima.ViewMode.NORMAL;
+          this.imaAdsManager_.resize(this.video_.offsetWidth,
+              this.video_.offsetHeight, viewMode);
+        });
+        this.resizeObserver_.observe(this.video_);
```

Comment:
Hm. We should also call `this.resizeObserver_.disconnect()` when the ad manager is disconnected from the video, so as to not leave hanging observers.

Unfortunately, now that I am looking at it, we don't have any appropriate lifecycle events for that. `AdManager.onAssetUnload` is called when the player unloads, but that's not the right moment; if you unload one asset and load another, the ad manager should still be there. We'd want to call it when the player detaches instead, I think, and that isn't passed on to the ad manager.

For now, I guess I'll approve and merge this current CL. I'll make a follow-up CL to make `Player.detach` also dispose of the ad manager, since it's kind of outside the scope of the issue you are fixing here.